// jsenv-brunch/lib/index.js
// Generated by CoffeeScript 1.4.0
(function() {
  var JsenvCompiler, coffee, sysPath, vm;

  vm = require('vm');

  coffee = require('coffee-script');

  sysPath = require('path');

  module.exports = JsenvCompiler = (function() {

    JsenvCompiler.prototype.brunchPlugin = true;

    JsenvCompiler.prototype.type = 'javascript';

    JsenvCompiler.prototype.pattern = /\.(js|coffee)env$/;

    function JsenvCompiler(config) {
      this.config = config;
      null;
    }

    JsenvCompiler.prototype.compile = function(data, path, callback) {
      var envHash, error, key, parsed, result, sandbox;
      sandbox = {
        module: {
          exports: void 0
        },
        require: require
      };
      try {
        if (sysPath.extname(path) === '.coffeeenv') {
          data = coffee.compile(data, {
            bare: true
          });
        }
        parsed = vm.runInNewContext("module.exports = " + data, sandbox);
        if (typeof parsed === "function") {
          envHash = parsed(process.env);
        } else {
          envHash = parsed;
          for (key in envHash) {
            if (process.env[key]) {
              envHash[key] = process.env[key];
            }
          }
        }
        return result = "module.exports = " + JSON.stringify(envHash);
      } catch (err) {
        return error = err;
      } finally {
        callback(error, result);
      }
    };

    return JsenvCompiler;

  })();

}).call(this);
